src/twisted/web/http.py

```diff
@@ -2250,6 +2250,9 @@ def lineReceived(self, line):
         Called for each line from request until the end of headers when
         it enters binary mode.
         """
+        assert (
+            not self._handlingRequest
+        ), "when handling a request, we MUST be in raw mode to buffer the incoming data without parsing it"
```

Comment:
I wanted the tests to run with this once but we can probably remove it if CI is happy.